arcgis-ios-sdk-samples/Layers/Browse OGC API feature service/BrowseOGCAPIServicesViewController.swift

```diff
@@ -0,0 +1,224 @@
+        // Remove observer on cancel.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { _ in
+            NotificationCenter.default.removeObserver(textFieldObserver!)
+            completion(.failure(InputError.userCancelled))
+        }
+        // Validate and create URL from user input and remove observer.
+        let loadAction = UIAlertAction(title: "Load", style: .default) { _ in
+            NotificationCenter.default.removeObserver(textFieldObserver!)
+            guard let text = alertController.textFields![0].text?.trimmingCharacters(in: .whitespacesAndNewlines),
+                  let serviceURL = URL(string: text) else {
+                completion(.failure(InputError.invalidURL))
+                return
+            }
+            completion(.success(serviceURL))
+        }
+        alertController.addAction(cancelAction)
+        alertController.addAction(loadAction)
+        alertController.preferredAction = loadAction
+        
+        // The text field for OGC API service URL.
+        alertController.addTextField { [self] textField in
+            textField.placeholder = defaultServiceURL.absoluteString
+            textField.text = defaultServiceURL.absoluteString
+            textField.keyboardType = .URL
+            textFieldObserver = NotificationCenter.default.addObserver(
+                forName: UITextField.textDidChangeNotification,
+                object: textField,
+                queue: .main
+            ) { [unowned loadAction] _ in
+                let text = textField.text!.trimmingCharacters(in: .whitespacesAndNewlines)
+                // Enable the load button if textfield is not empty.
+                loadAction.isEnabled = !text.isEmpty
+            }
+        }
+        present(alertController, animated: true)
+    }
+    
+    @IBAction func browseLayerInfos(_ sender: UIBarButtonItem) {
+        let selectedIndex = featureCollectionInfos.firstIndex { $0 == selectedInfo }
+        let optionsViewController = OptionsTableViewController(labels: featureCollectionInfos.map { $0.title }, selectedIndex: selectedIndex) { [self] newIndex in
+            let selectedInfo = featureCollectionInfos[newIndex]
+            self.selectedInfo = selectedInfo
+            loadLayer(info: selectedInfo)
+        }
+        optionsViewController.modalPresentationStyle = .popover
+        optionsViewController.preferredContentSize = CGSize(width: 300, height: 300)
+        optionsViewController.presentationController?.delegate = self
+        optionsViewController.popoverPresentationController?.barButtonItem = sender
+        present(optionsViewController, animated: true)
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["BrowseOGCAPIFeatureServiceViewController"]
```

Comment:
I cannot view the source code in the sample viewer. I think the file name is off here.